server/auth/router.js

```diff
@@ -15,8 +17,39 @@ const createAuthToken = function(user) {
   });
 };
 
-const localAuth = passport.authenticate("local", { session: false });
 router.use(bodyParser.json());
+
+router.post("/socialLogin", (req, res) => {
+  let { username, password = username, name } = req.body;
```

Comment:
I think you should also add a parameter to distinguish FB users from other social networks to avoid possible collisions.
Also passing username inside password field though seems to be safe it is confusing.
I recommend supporting User.create without mandatory password later on